centreon/www/install/steps/process/migrateCredentialsToVault.php

```diff
@@ -0,0 +1,289 @@
+ * @param CentreonRestHttp $httpClient
+ *
+ * @return string
+ *
+ * @throws RestBadRequestException
+ * @throws RestConflictException
+ * @throws RestForbiddenException
+ * @throws RestInternalServerErrorException
+ * @throws RestMethodNotAllowedException
+ * @throws RestNotFoundException
+ * @throws RestUnauthorizedException
+ */
+function migrateDatabaseCredentials(
+    VaultConfiguration $vaultConfiguration,
+    string $token,
+    CentreonRestHttp $httpClient
+): string {
+    $uuidGenerator = new \Utility\UUIDGenerator();
+    $uuid = $uuidGenerator->generateV4();
+    $vaultPathUri = "jeremy/data/database/" . $uuid;
+    $vaultPath = "secret::hashicorp_vault::" . $vaultPathUri;
+    $credentials = retrieveDatabaseCredentialsFromConfigFile();
+    $url = 'https://' . $vaultConfiguration->getAddress() . ':' . $vaultConfiguration->getPort()
+        . '/v1/' . $vaultPathUri;
+    $headers = [
+        'X-Vault-Token: ' . $token,
+    ];
+
+    $body = [
+        'data' => ['_DBUSERNAME' => $credentials['username'], '_DBPASSWORD' => $credentials['password']],
+    ];
+
+    $httpClient->call($url, 'POST', $body, $headers);
+
+    return $vaultPath;
+}
+
+/**
+ * Update the different config files with the vault path.
+ *
+ * @param $vaultPath
+ *
+ * @throws Exception
+ */
+function updateConfigFilesWithVaultPath($vaultPath): void
+{
+    updateCentreonConfPhpFile($vaultPath);
+    updateCentreonConfPmFile($vaultPath);
+    updateDatabaseYamlFile($vaultPath);
+}
+
+/**
+ * Retrieve database credentials from config file.
+ *
+ * @return array{username: string, password: string
+ *
+ * @throws Exception
+ */
+function retrieveDatabaseCredentialsFromConfigFile(): array
+{
+    $content = file_get_contents("/etc/centreon/centreon.conf.php")
+        ?: throw new Exception('Unable to retrieve content of file');
+
+    preg_match(
+        '/\$conf_centreon\[[\'\"]user[\'\"]\]\s*=\s*[\'\"](.*)[\'\"]\s*;/',
+        $content,
+        $matches
+    );
+
+    $userContent = $matches[1];
+
+    preg_match(
+        '/\$conf_centreon\[[\'\"]password[\'\"]\]\s*=\s*[\'\"](.*)[\'\"]\s*;/',
+        $content,
+        $matches
+    );
+
+    $passwordContent = $matches[1];
+
+    return ['username' => $userContent, 'password' => $passwordContent];
+}
+
+/**
+ * @param string $vaultPath
+ *
+ * @throws Exception
+ */
+function updateCentreonConfPhpFile(string $vaultPath): void
+{
+    $content = file_get_contents("/etc/centreon/centreon.conf.php")
+        ?: throw new Exception('Unable to retrieve content of file: /etc/centreon/centreon.conf.php');
+
+    $newContentPhp = preg_replace(
+        '/\$conf_centreon\[[\'\"]user[\'\"]\]\s*=\s*(.*)/',
+        '\$conf_centreon[\'user\'] = \'' . $vaultPath . '\';',
+        $content
+    );
+    $newContentPhp = preg_replace(
+        '/\$conf_centreon\[[\'\"]password[\'\"]\]\s*=\s*(.*)/',
+        '\$conf_centreon[\'password\'] = \'' . $vaultPath . '\';',
+        $newContentPhp
+    );
```

Comment:
You must not modify the variables in this file.
Make the necessary changes so that /usr/share/centron/centreon.conf.php redefines the constants according to the Feature Flag 'Vault'.

If the problem concerns the use of these variables in the code, then replace them with the use of constants

Reply:
This is all the intends of the tickets, we do not want the credentials in raw value in those 3 files. So we update the files with the vault path instead of the credentials in raw value, then the centreon.config.php already have a mechanism to retrieve true credentials from vault.